examples/outer-join-clause/outer_join_clause.bal

```diff
@@ -0,0 +1,42 @@
+import ballerina/io;
+
+type Department record {|
+   int id;
+   string name;
+|};
+
+type Person record {|
+   int id;
+   string fname;
+   string lname;
+|};
+
+type DeptPerson record {|
+   string fname;
+   string? dept;
+|};
+
+public function main() {
+    Person p1 = {id: 1, fname: "Alex", lname: "George"};
+    Person p2 = {id: 2, fname: "John", lname: "Fonseka"};
+    Person p3 = {id: 3, fname: "Ted", lname: "Perera"};
+
+    Department d1 = {id: 1, name:"HR"};
+    Department d2 = {id: 2, name:"Operations"};
+
+    Person[] personList = [p1, p2, p3];
+    Department[] deptList = [d1, d2];
+
+    DeptPerson[] deptPersonList =
+       from var person in personList
+       // The `outer join` clause include all of the records from the left table,
+       // even if there are no matching values for records in the right table.
+       outer join Department dept in deptList
```

Comment:
Btw, this is will be disallowed with https://github.com/ballerina-platform/ballerina-lang/issues/38223